src/logic/safe/store/reducer/gatewayTransactions.ts

```diff
@@ -77,6 +78,16 @@ export const gatewayTransactionsReducer = handleActions<GatewayTransactionsState
             // pushing a newer transaction to the existing list messes the transactions order
             // this happens when most recent transactions are added to the existing txs in the store
             newHistory[startOfDate] = newHistory[startOfDate].sort((a, b) => b.timestamp - a.timestamp)
+
+            // remove tx from queued if it exists there
+            if (isMultisigExecutionInfo(transaction.executionInfo)) {
+              console.log(transaction.executionInfo.nonce)
+              const txNonce = transaction.executionInfo.nonce
+
+              if (newQueued?.queued && newQueued?.queued[txNonce]) {
```

Comment:
It's safest to also check the `'queued.next'` location as well.

Reply:
You could reduce this code by means of optional chaining:

```ts
  delete newQueued?.queued?.[txNonce]
```

Reply:
true! Thanks for the correction